Update open-source-services.md - opensourcing tests from legal perspective (#149)
operate-first/community: open-source-services.md

**A:** Tests that are used during operation the service (as per **requirement #1**), and tests that used when run when making a change to the service (as per **requirement #2**), _must be shared publicly_ under an Open Source license.

> Tests that violate _legal restrictions_ cannot be shared publicly. Specifically: tests that include references to internal or customer names, urls, addresses, configuration or other data; tests for CVE, embargoed or private bugs; testing performance, failover, data recovery and compliance with security standards (like FIPS). 

Reviewer: Can we replace "internal" with "company private". 

Why do we list "tests for CVE" separate here? Doesn't embargoed cover the topic? Don't we want tests that verify a CVE fix to be available publically once that CVE is public and no longer embargoed, right?

Why do testing performance, failover, data recovery and compliance with standards tests violate _legal restrictions_?

Reviewer 2: +1 company private sounds good.
Do we at Red Hat have something like these tier levels as Top Secret, Secret, Confidential, as I see in some presentations? Maybe we could use that level to define what can be shared and what not?

Reviewer 3: This is a fix just for what seems clear here so far:

```suggestion
Tests that violate _legal restrictions_ cannot be shared publicly.
Specifically: tests that include references to company and customer private information, URLs, IP addresses, configuration or other data; tests for CVE, embargoed or private bugs; testing performance, failover, data recovery and compliance with security standards (like FIPS). 
```

Reviewer 3: > Can we replace "internal" with "company private".

Addressed in my suggestion.

> Why do we list "tests for CVE" separate here? Doesn't embargoed cover the topic? Don't we want tests that verify a CVE fix to be available publically once that CVE is public and no longer embargoed, right?

I agree there is nuance here that includes some level and time of openness; perhaps just "ermbargoed or private bugs" covers all the cases, and we can work out the specific details within a security SIG.

> Why do testing performance, failover, data recovery and compliance with standards tests violate _legal restrictions_?

Good question, I can make some guesses, but we need some fairly concrete explanations of why something is kept private/secret.

Author: I am reaching to expert in legal team to provide answer, stay tuned.

Reviewer 4: I got response from RH legal team I'll reply comments and incorporate the changes by April 20.

Author: Comments and commits incorporated, please review again

> It is _recommended_ that other tests which are independent of the service and (eg: penetration tests) are Open Source, but this is not required.

Reviewer 2: +1

Reviewer 3: ```suggestion
It is _recommended_ other tests that are independent of the service (e.g. penetration tests) are Open Source, but this is not required.
```
